src/Analyzers/Core/CodeFixes/UseNullPropagation/AbstractUseNullPropagationCodeFixProvider.cs

```diff
@@ -153,6 +156,24 @@ public override Task RegisterCodeFixesAsync(CodeFixContext context)
                 syntaxFacts, generator, whenPartIsNullable, whenTrueStatement, match);
             Contract.ThrowIfNull(newWhenTrueStatement);
 
+            // If we have code like:
+            // ...
+            // else if (v != null)
+            // {
+            //     v.M();
+            // }
+            // then we want to keep the result statement in a block:
+            // else
+            // {
+            //     v?.M();
+            // }
+            // Applies only to C# since VB doesn't have a general-purpose block syntax
+            if (syntaxFacts.IsElseClause(originalIfStatement.Parent) &&
+                IsBlockSyntax(whenTrueStatement.Parent))
+            {
+                newWhenTrueStatement = Block(newWhenTrueStatement).WithAdditionalAnnotations(Formatter.Annotation);
```

Comment:
It's not recommended to use formatter in codefixes as far as I know.

Can you check if the fix could be done in `CreateConditionalAccessExpression` to not lose the block in the first place?

Reply:
Well, by reusing the block I _have to_ format it as well. Otherwise the result is:
```cs
else {
    statement here
}
```
I can technically prepend trivia form `if`'s `)` to get the new line, but it's very much C#-only and requres quite a lot of code around it. I think, having to format a single block with a single statement in it isn't _that_ big problem.

@CyrusNajmabadi what do you think about it?

Reply:
@DoctorKrolic We should preserve the formatting the user had. If the formatting was incorrect before applying the codefix, we should keep it incorrect.

Reply:
No, I mean we get such formatting from perfectly good code before:
```cs
else if (...)
{
    ...
}
```

Reply:
@DoctorKrolic What if the good wasn't formatted properly before?

Reply:
The formatter takes into consideration user's preference about whether to put `{` on the same line or not. If there was `else if (...) {` while the rule is to put `{` on the next line, it would be formatted according to this rule:
```cs
else
{
}
```

Yes, it might be good to keep formatting "as is", but I am not sure if it worth the cost. Again, this is very much C#-only problem, so in order to determine, whether we want a line break or not, I'll have to create an abstract method with C#-only implementation and just throwing in VB etc.

Reply:
> I'll have to create an abstract method with C#-only implementation and just throwing in VB etc.

Isn't it just a matter of copying the trailing trivia from the condition?

Reply:
Well, this doesn't seem possible to me (unless there is a way to explicitly forbid formatter to format specific node). If you debug current state of the PR, you would be able to see, that I preserve all trivia "as is" now. However, there is a cleanup phase after the codefix and formatting is one of the stages there. And formatter forces `else {` to become multiline...

Reply:
Formatter should only touch nodes with an formatting annotation on them, or which have elastic trivia.  That said, I don't care much about being ultra correct here.

Reply:
Never mind, I just forgot to remove adding formatter annotation to the block...